### PR TITLE
Buildroot version 2018.02 [DEVC-413]

### PR DESCRIPTION
**Pull request by silverjam**
Switch to buildroot version 2018.02, which gives us support for a couple things, namely Rust and SDK generation support.  We use the SDK support to generate an SDK that can compile binaries for a given piksi buildroot built.  SDK packages are uploaded to S3 for any tagged build.

This version of buildroot changes some implicit package dependencies that we previously depended on... one in particular was libcurl, which is not built with SSL by default.

This change "harmonizes" the toolchains we are using, we're using the Buildroot toolchains from Bootlin (obtained here: https://toolchains.bootlin.com/)-- we're using the ARM toolchain for Piksi, and the x86_64 toolchain to build our unit tests.  The x86_64 is also used to build the LLVM toolchain for obfuscation.

### CODE REVIEW

**Comment by silverjam** [comment]
Looks like there's still issues with the host build and the SDK build, will investigate at some point.

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018042720-4-ge5ad45a
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=509171)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-4-ge5ad45a&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-4-ge5ad45a&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018042720-4-ge5ad45a
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018042720-4-ge5ad45a)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018042720-4-ge5ad45a/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018042720-4-ge5ad45a

**Comment by silverjam** [comment]
OK, only SDK/ip-security build is failing now...

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018042720-9-gaad6310
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=514012)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-9-gaad6310&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-9-gaad6310&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018042720-9-gaad6310
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018042720-9-gaad6310)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018042720-9-gaad6310/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018042720-9-gaad6310

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018042720-10-ge11f488
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=514052)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-10-ge11f488&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-10-ge11f488&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018042720-10-ge11f488
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018042720-10-ge11f488)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018042720-10-ge11f488/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018042720-10-ge11f488

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018042720-13-g51120ed
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=514177)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-13-g51120ed&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-13-g51120ed&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018042720-13-g51120ed
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018042720-13-g51120ed)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018042720-13-g51120ed/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018042720-13-g51120ed

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018042720-14-ge1a0e85
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=515333)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-14-ge1a0e85&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018042720-14-ge1a0e85&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018042720-14-ge1a0e85
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018042720-14-ge1a0e85)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018042720-14-ge1a0e85/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018042720-14-ge1a0e85

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018050419-15-g9570108
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=522160)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-15-g9570108&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-15-g9570108&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018050419-15-g9570108
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018050419-15-g9570108)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018050419-15-g9570108/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018050419-15-g9570108

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018050419-16-g5deb09e
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=522227)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-16-g5deb09e&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-16-g5deb09e&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018050419-16-g5deb09e
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018050419-16-g5deb09e)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018050419-16-g5deb09e/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018050419-16-g5deb09e

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018050419-22-gbe63bba
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=523036)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-22-gbe63bba&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-22-gbe63bba&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018050419-22-gbe63bba
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018050419-22-gbe63bba)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018050419-22-gbe63bba/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018050419-22-gbe63bba

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018050419-23-g97d30b8
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=523670)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-23-g97d30b8&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-23-g97d30b8&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018050419-23-g97d30b8
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018050419-23-g97d30b8)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018050419-23-g97d30b8/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018050419-23-g97d30b8

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018050419-25-g87200c8
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=524063)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-25-g87200c8&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-25-g87200c8&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018050419-25-g87200c8
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018050419-25-g87200c8)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018050419-25-g87200c8/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018050419-25-g87200c8

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018050419-27-g0a4c224
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=524119)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-27-g0a4c224&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-27-g0a4c224&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018050419-27-g0a4c224
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018050419-27-g0a4c224)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018050419-27-g0a4c224/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018050419-27-g0a4c224

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018050419-43-g30e52da
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=525959)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-43-g30e52da&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018050419-43-g30e52da&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018050419-43-g30e52da
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018050419-43-g30e52da)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018050419-43-g30e52da/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018050419-43-g30e52da

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018051217-31-g3487bf0
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=526781)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018051217-31-g3487bf0&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018051217-31-g3487bf0&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018051217-31-g3487bf0
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018051217-31-g3487bf0)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018051217-31-g3487bf0/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018051217-31-g3487bf0

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018051217-33-gfb844de
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=526907)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018051217-33-gfb844de&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018051217-33-gfb844de&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018051217-33-gfb844de
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018051217-33-gfb844de)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018051217-33-gfb844de/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018051217-33-gfb844de

File: configs/piksiv3_defconfig [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+BR2_TOOLCHAIN_EXTERNAL_URL="http://toolchains.free-electrons.com/downloads/releases/toolchains/armv7-eabihf/tarballs/armv7-eabihf--glibc--stable-2018.02-2.tar.bz2"
+BR2_TOOLCHAIN_EXTERNAL_GCC_6=y
+BR2_TOOLCHAIN_EXTERNAL_HEADERS_4_1=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_GLIBC=y
+BR2_TOOLCHAIN_EXTERNAL_CXX=y
+BR2_TOOLCHAIN_EXTERNAL_GDB_SERVER_COPY=y
+BR2_TOOLCHAIN_GLIBC_GCONV_LIBS_COPY=y
 BR2_TARGET_OPTIMIZATION="-flto"
-BR2_TARGET_CFLAGS="-flto"
 BR2_TARGET_LDFLAGS="-flto"
```
Comment (benjaminaltieri) [inline configs/piksiv3_defconfig]: Is this redundant to the line in lto.mk that also sets -flto on BR2_TARGET_LDFLAGS?

Comment (silverjam) [reply configs/piksiv3_defconfig]: Yes, it should be

Comment (silverjam) [reply configs/piksiv3_defconfig]: I haven't nailed down all the interactions with the flags/variables in buildroot, but it seems the CFLAGS one isn't used (or isn't recognized) by the buildroot config system, so when `make menuconfig` is run or buildroot it drops the `BR2_TARGET_CFLAGS` variable.

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018051217-7-gd7ef023
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=528201)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018051217-7-gd7ef023&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018051217-7-gd7ef023&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018051217-7-gd7ef023
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018051217-7-gd7ef023)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018051217-7-gd7ef023/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018051217-7-gd7ef023

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.4.0-develop-2018051217-7-g025e810
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=528252)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018051217-7-g025e810&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.4.0-develop-2018051217-7-g025e810&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.4.0-develop-2018051217-7-g025e810
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.4.0-develop-2018051217-7-g025e810)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.4.0-develop-2018051217-7-g025e810/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.4.0-develop-2018051217-7-g025e810

**Pull request merged**